Source/Core/Core/State.cpp

```diff
@@ -354,6 +306,58 @@ static std::map<double, int> GetSavedStates()
   return m;
 }
 
+static void CompressBufferToFile(const u8* raw_buffer, u32 size, File::IOFile& f)
+{
+  u32 total_bytes_compressed = 0;
```

Comment:
`size`, `total_bytes_compressed`, `bytes_left_to_compress` should be `u64`, there's no reason to limit that to 4GB.